File: src/Entity/Message.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use JMS\Serializer\Annotation\Groups;


class Message {
	/**
	 * @var string
	 *
	 * @Groups({"product"})
	 */
	private $action;

	/**
	 * @var Product
	 *
	 * @Groups({"product"})
	 */
	private $result;

	/**
	 * @var Product[]
	 *
	 * @Groups({"product"})
	 */
	private $results;

	/**
	 * @return string
	 */
	public function getAction() {
		return $this->action;
	}

	/**
	 * @param string $action
	 * @return Message
	 */
	public function setAction($action) {
		$this->action = $action;

		return $this;
	}

	/**
	 * @return Product
	 */
	public function getResult() {
		return $this->result;
	}

	/**
	 * @param Product $result
	 * @return Message
	 */
	public function setResult($result) {
		$this->result = $result;

		return $this;
	}

	/**
	 * @return Product[]
	 */
	public function getResults() {
		return $this->results;
	}

	/**
	 * @param Product[] $results
	 * @return Message
	 */
	public function setResults($results) {
		$this->results = $results;

		return $this;
	}

}

File: java-agent/src/main/java/fr/pharmaciepouvreau/inventaire/dto/Message.java
package fr.pharmaciepouvreau.inventaire.dto;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Message<T> implements Serializable {
	private static final long serialVersionUID = 5813063486110672710L;

	private String description;

	private String action;

	private List<T> results = new ArrayList<T>();

	private T result;

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getAction() {
		return action;
	}

	public void setAction(String action) {
		this.action = action;
	}

	public List<T> getResults() {
		return results;
	}

	public void setResults(List<T> results) {
		this.results = results;
	}

	public void addResult(T result) {
		this.results.add(result);
	}

	public T getResult() {
		return result;
	}

	public void setResult(T result) {
		this.result = result;
	}

}
